### DISCUSSION TITLE
Running eth_mnist.py on GPU(cuda), but it took approximately 14 hours, am I doing something incorrect?

### DISCUSSION THREAD

**Discussion by ghost** [BindsNET/bindsnet#456] [Q&A] [answered]
Good evening, 
I am new in Spiking Neural Networks(overall in DL/ML), It is my first startup. 
I've started it with BindsNET. I tried to run eth_mnist.py, but it took almost 14 hours. Could you tell me please am I doing something wrong?
Here I'm attaching my device information.
I am using WSL2
[
![gpu](https://user-images.githubusercontent.com/75060403/107933791-92132980-6fc2-11eb-8a22-6e8e35c07b3e.PNG)
](url)

**Comment by Hananel-Hazan** [comment, marked as answer]
I am assuming that you are using Windows 10 under the Dev channel, therefore, the pre-release beta version WSL2 has GPU access.  WSL2 added much-needed features compared to WSL1 like GPU access. But where WSL2 can access the GPU with no issue,  performance-wise, it leaves much room for improvements (therefore, it beta status). The performance didn't improve much since June 2020 [link ](https://www.phoronix.com/scan.php?page=news_item&px=WSL2-CUDA-Perf-Early-Look)

Tip, when you run PyTorch task that use GPU under WSL2, check the Windows 10, Task Manager - performance tab, you can see that your GPU is underutilized.

If you want to utilize your GPU, you have two options. 1) use native Linux - the preferred approach. 2) use native python on windows 10 with GPU access. (I never used this approach, so I don't know if it is working well.)

PS, if you run the program without plots, it will but much faster

**Reply by ghost** [reply]
Good evening, Yes, I'm using Windows 10 under the Dev channel. Thank you for your answer. 
I will check the Tip.
